### ISSUE TITLE
Can't use case/when based on measures

### ISSUE THREAD

**Issue by ifokeev** [cube-js/cube.js#260]
**Describe the bug**
I have the schema:

```
cube(`Trades`, {
  ...
  
  measures: {
    ...

    buysCount: {
      sql: `${CUBE}."id"`,
      type: 'count',
      filters: [
        { sql: `${CUBE}."side" = 'buy'` }
      ]
    },

    sellsCount: {
      sql: `${CUBE}."id"`,
      type: 'count',
      filters: [
        { sql: `${CUBE}."side" = 'sell'` }
      ]
    },
  },
  
  dimensions: {
    signal: {
      type: `string`,
      case: {
        when: [
            { sql: `${buysCount} >= 50`, label: `1` },
            { sql: `${sellsCount} >= 50`, label: `2` },
        ],
        else: { label: `0` }
      }
    }
  }
});
```

And I can't use these measures in `case when`, bacause cubejs generating SQL like:

```
 SELECT
       CASE
         WHEN count(CASE WHEN (trades."side" = 'buy') THEN trades."id" END) >= 50 THEN '1'
         WHEN count(CASE WHEN (trades."side" = 'sell') THEN trades."id" END) >= 50 THEN '2' 
         ELSE '0' END "trades.signal",
         count(CASE WHEN (trades."side" = 'buy') THEN trades."id" END) "trades.buys_count", 
         count(CASE WHEN (trades."side" = 'sell') THEN trades."id" END) "trades.sells_count"
     FROM
       (
         ...mainSQL
       ) AS trades
   GROUP BY 1 LIMIT 5000
```

It adds `GROUP BY` statement by default for dimensions and the error here: 
`Error: aggregate functions are not allowed in GROUP BY`

**Expected behavior**
Without `GROUP BY 1` my query works as expected. May be it's ok to add options to skip grouping for dimensions or allow `case when` in measures

**Version:**
[e.g. 0.11.18]

**Comment by paveltiunov** [comment]
@ifokeev Hey Ivan! Yep. That's right. In your case `signal` should be a measure instead of dimension. Dimension can reference measure only in case of `subQuery`. I see two solutions:
1. Write this case in sql of a `signal` measure as a number.
2. Introduce `case` support for measures.

We can use this issue to track last one. What do you think?

**Comment by ifokeev** [comment]
@paveltiunov in case it will be a `measure` then label only could be as `number`, but it's not true. Label in `case when` may be string or dynamic https://cube.dev/docs/dimensions#parameters-case, and it's not a real `measure` here. I think about this `signal` as a dimension, so the right way here is just to control grouping: 
– fix `GROUP BY` for `case when` when it contains measures

or add something specific, for example, `skipGrouping`.

**Comment by paveltiunov** [comment]
@ifokeev Yep. It's pretty tricky. Actually measures can be strings, booleans or any other type as soon as there's a rule for aggregation. There're simple rules about measures vs dimensions:
1. Measures can be referenced only by measures
2. Dimensions can be referenced by both dimensions and measures.
3. If dimension is referenced by measure it should be aggregated then.
4. Exceptional case of referencing measure by dimension is `subQuery`.
You can find more info about it in Appendix E of http://nnhoa.free.fr/courses/OLAP/olap-solutions-building-multidimensional-information-systems978047140030125538.pdf

I believe support of `case` in measure and `string` measure type should solve your issue.

**Comment by ifokeev** [comment]
@paveltiunov thank you a lot for this book, I was searching for a good OLAP explanation.
So, you're right. In terms of `variables` and `dimensions`, it's a `variable` that should be the `string`.

**Comment by jhofeditz** [comment]
It would be very nice for CubeJS to support string as an allowed type for measures. I ran into this attempting to add an aggregate string as a measure using string_agg in Postgres. I think my workaround for now will be to create a view to perform the string_agg.
